### ISSUE TITLE
AttributeError: Can't pickle local object 'persist_messages.<locals>.consumer'

### ISSUE THREAD

**Issue by pnadolny13** [estrategiahq/target-parquet#10]
I'm running on a Mac (Version 11.6.8, i9). I'm getting the following error related to multi processing forking:

```
Traceback (most recent call last):
  File "/Users/pnadolny/Documents/Git/GitHub/pnadolny/hub-utils/hub_utils/test_meltano_project/.meltano/loaders/target-parquet/venv/bin/target-parquet", line 8, in <module>
    sys.exit(main())
  File "/Users/pnadolny/Documents/Git/GitHub/pnadolny/hub-utils/hub_utils/test_meltano_project/.meltano/loaders/target-parquet/venv/lib/python3.9/site-packages/target_parquet/__init__.py", line 274, in main
    state = persist_messages(
  File "/Users/pnadolny/Documents/Git/GitHub/pnadolny/hub-utils/hub_utils/test_meltano_project/.meltano/loaders/target-parquet/venv/lib/python3.9/site-packages/target_parquet/__init__.py", line 225, in persist_messages
    t2.start()
  File "/Users/pnadolny/.pyenv/versions/3.9.7/lib/python3.9/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/pnadolny/.pyenv/versions/3.9.7/lib/python3.9/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/Users/pnadolny/.pyenv/versions/3.9.7/lib/python3.9/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/Users/pnadolny/.pyenv/versions/3.9.7/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Users/pnadolny/.pyenv/versions/3.9.7/lib/python3.9/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/pnadolny/.pyenv/versions/3.9.7/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/Users/pnadolny/.pyenv/versions/3.9.7/lib/python3.9/multiprocessing/reduction.py", line 61, in dump
    ForkingPickler(file, protocol).dump(obj)
AttributeError: Can't pickle local object 'persist_messages.<locals>.consumer'
```

Which landed me on https://stackoverflow.com/questions/65834254/multiprocessing-runtime-error-freeze-support-in-mac-64-bit and linked me to this blog https://pythonspeed.com/articles/python-multiprocessing/ explaining issues with forking and Macs.

I confirmed that I could run this using the default python:3.9 docker image which is linux based, so its definitely my Mac.

I'm not super familiar with multiprocessing, do you know of a way to resolve this so the tap support Mac again (assuming I'm not the only person with this issue, but maybe)?

**Comment by bmritz** [comment]
I have seen this issue as well on macOS.

**Comment by acarter24** [comment]
Seeing this too on Windows 11, py3.9 , meltano 2.15.3

**Comment by edgarrmondragon** [comment]
I think I have a fix for this using [the `fork` method](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods). I'll fork the repo and try it out.

**Comment by auyer** [comment]
This seems to have broken Linux compatibility. Can you give it a look @edgarrmondragon ?

**Comment by EMCP** [comment]
I think it's patched for ubuntu/linux now via #16
